libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/MultiLanguageGeneratorBase.cs

```diff
@@ -40,20 +42,43 @@ public MultiLanguageGeneratorBase()
         /// <returns>The generator.</returns>
         public override async Task<string> Generate(ITurnContext turnContext, string template, object data)
         {
-            // see if we have any locales that match
             var targetLocale = turnContext.Activity.Locale?.ToLower() ?? string.Empty;
 
-            var locales = new string[] { string.Empty };
-            if (!this.LanguagePolicy.TryGetValue(targetLocale, out locales))
+            // priority 
+            // 1. local policy
+            // 2. shared policy in turnContext
+            // 3. default policy
+            var languagePolicy = this.LanguagePolicy ?? 
+                                turnContext.TurnState.Get<LanguagePolicy>() ?? 
+                                new LanguagePolicy();
+
+            // see if we have any locales that match, if not, use empty as fallback
+            var fallbackLocales = new List<string>();
+            if (languagePolicy.ContainsKey(targetLocale))
+            {
+                fallbackLocales.AddRange(languagePolicy[targetLocale]);
+            }
+            else
+            {
+                fallbackLocales.AddRange(languagePolicy[string.Empty]);
+            }
+
+            if (fallbackLocales.Count == 0)
+            {
+                throw new Exception($"No supported language found for {targetLocale}");
+            }
+            else
             {
-                if (!this.LanguagePolicy.TryGetValue(string.Empty, out locales))
+                // if last locale point to "", we do one more re-direct
+                if (fallbackLocales.Last() == string.Empty)
                 {
-                    throw new Exception($"No supported language found for {targetLocale}");
+                    fallbackLocales.RemoveAt(fallbackLocales.Count - 1);
+                    fallbackLocales.AddRange(languagePolicy[string.Empty]);
                 }
```

Comment:
I would not do this:
Change LanguagePolicy.cs to pass in default language

    /// <summary>
    /// Language policy with fallback for each language as most specific to default en-us -> en -> default.
    /// </summary>
    public class LanguagePolicy : Dictionary<string, string[]>
    {
        // walk through all of the cultures and create a dictionary map with most specific to least specific
        // Example output "en-us" will generate fallback rule like this:
        //   "en-us" -> "en" -> "" 
        //   "en" -> ""
        // So that when we get a locale such as en-gb, we can try to resolve to "en-gb" then "en" then ""
        // See commented section for full sample of output of this function
        public LanguagePolicy(string defaultLanguage = null)
            : base(StringComparer.OrdinalIgnoreCase)
        {
            if (defaultLanguage == null)
            {
                defaultLanguage = string.Empty;
            }

            var cultureCodes = CultureInfo.GetCultures(CultureTypes.AllCultures).Select(c => c.IetfLanguageTag.ToLower()).ToList();
            foreach (var language in cultureCodes.Distinct())
            {
                var lang = language.ToLower();
                var fallback = new List<string>();
                while (!string.IsNullOrEmpty(lang))
                {
                    fallback.Add(lang);

                    var i = lang.LastIndexOf("-");
                    if (i > 0)
                    {
                        lang = lang.Substring(0, i);
                    }
                    else
                    {
                        break;
                    }
                }

                if (!fallback.Contains(defaultLanguage))
                {
                    fallback.Add(defaultLanguage);
                }

                this.Add(language, fallback.ToArray());
            }
        }
    }

Reply:
This is not functional equals to what i'm doing there, because this only allow one certain default language. You can see my test cases here i set "" => ["fr-fr", ""], which gives me the capability of having fr-fr as default language if target locale not exist, then if foo.fr-fr.lg not exist, i will again to check foo.lg.   

Besides the difference in functionality, this also don't adds any additional concept like "defaultLanguage" (where will serialize this?), we are only dealing with the policy. 

If we can achieve more with less concept, why not just go for it, right?